### PR TITLE
General: Use yarn for more building

### PR DESCRIPTION
**Pull request by kraftbj**
The `build-jetpack.sh` file is used by the ~beta builder server~, but it calls only the gulp build actions.

#### Changes proposed in this Pull Request:
* Use yarn scripts everywhere for building to try to have an uniform experience of the build.

#### Testing instructions:
* Does it build using either yarn build and the build-jetpack.sh script?

#### Proposed changelog entry for your changes:
n/a

*

### CODE REVIEW

**Review by brbrr — DISMISSED** [review]
Seems like this PR did not fix the problem for me. I still see un-built artifacts on Jetpack dashboard and Jetpack blocks are missing. I've launched a JN site with `https://jurassic.ninja/create/?jetpack-beta&branch=update/build-script-to-yarn&shortlived`

File: package.json [hunk 1]
```diff
@@ -24,8 +24,8 @@
     "build": "yarn install-if-deps-outdated && yarn clean-client && yarn build-client && yarn build-extensions",
     "build-client": "gulp",
     "build-extensions": "webpack --config ./webpack.config.extensions.js",
-    "build-production-client": "NODE_ENV=production BABEL_ENV=production yarn build-client",
-    "build-production": "yarn distclean && yarn && gulp languages:extract && yarn build-production-client",
```
Comment (simison) [inline package.json]: Seems like we could hold off from these changes in this PR and just sort out the bash scripts out first.

Comment (kraftbj) [reply package.json]: The second line (gulp:language) needs to change since it expects to have the POT file from a built version in my testing.

**Comment by kraftbj** [comment]
Holding off on this until the dust settled from the node changes on the beta build and to better identify where the bash script is used.

File: tools/build-jetpack.sh [hunk 2]
```diff
@@ -61,9 +61,8 @@ hash yarn 2>/dev/null || {
     echo >&2 "Please install it following the instructions on https://yarnpkg.com. Aborting.";
     exit 1;
 }
-
-yarn --modules-folder=$TARGET_DIR/node_modules
-NODE_ENV=production BABEL_ENV=production $TARGET_DIR/node_modules/.bin/gulp
+yarn --cwd $TARGET_DIR cache clean
+yarn --cwd $TARGET_DIR run build-production
```
Comment (oskosk) [inline tools/build-jetpack.sh]: This may need to not be `build-production` but just `build`

Comment (zinigor) [inline tools/build-jetpack.sh]: This is cool, thank you! You can test if this works by going through testing instructions for #6308. Basically this script is needed if you want to have a built version of Jetpack clean from anything development related.

File: tools/build-release-branch.sh [hunk 3]
```diff
@@ -211,15 +211,9 @@ hash yarn 2>/dev/null || {
...
-yarn run distclean
-yarn run clean-client
-# Clean yarn's cache
+# Start cleaning the cache.
 yarn cache clean
-# Install dependencies
-yarn
-# Build the Admin Page
-NODE_ENV=production yarn run build-client
+yarn run build-production
```
Comment (sirreal) [inline tools/build-release-branch.sh]: Includes `distclean` 👍

**Review by sirreal — COMMENTED** [review]
I did not test, but the changes make sense.

**Comment by mdawaffe** [comment]
> Download the Beta built version of this branch (or use JN) - Do you have blocks? Does the admin dashboard work?

Tested in JN. Saw working dashboard and working blocks :)

> Do the same on current master and see the broken dashboard styles and no blocks.

Tested in JN. Saw working dashboard and working blocks :(

So I'm looking at the wrong thing and am not sure how to reproduce the original problem.

**Comment by kraftbj** [comment]
> So I'm looking at the wrong thing and am not sure how to reproduce the original problem.

That's my bad for not updating the original description and not realizing it. It was from when I thought the beta builder problems were due to the script, not the builder server node version.

Does it build using either `yarn build` and the build-jetpack.sh script?

**Review by mdawaffe — APPROVED** [review]
`yarn build` failed for me under node 11, but I guess that's fixed as of https://github.com/Automattic/wp-calypso/pull/31835.

`yarn build` worked for me under node 10. (Though it took a long time - I didn't time it). Blocks are present. Dashboard works.

`./tools/build-jetpack.sh -d -b update/build-script-to-yarn` worked for me under node 10. That is, it finished with no errors. I didn't check the output :) It took about 10 minutes.

**Comment by kraftbj** [comment]
That is longer than normal, but I want to explore that under a different PR since the time isn't the concern of the issue this is fixing.

**Pull request merged**